stdlib/unittest/runner.pyi

```diff
@@ -14,23 +15,55 @@ class TextTestResult(unittest.result.TestResult):
     separator2: str
     showAll: bool  # undocumented
     stream: TextIO  # undocumented
-    def __init__(self, stream: TextIO, descriptions: bool, verbosity: int) -> None: ...
+    if sys.version_info >= (3, 12):
+        durations: unittest.result._DurationsType | None
+        def __init__(
+            self, stream: TextIO, descriptions: bool, verbosity: int, *, durations: unittest.result._DurationsType | None = None
+        ) -> None: ...
+    else:
+        def __init__(self, stream: TextIO, descriptions: bool, verbosity: int) -> None: ...
+
     def getDescription(self, test: unittest.case.TestCase) -> str: ...
     def printErrorList(self, flavour: str, errors: Iterable[tuple[unittest.case.TestCase, str]]) -> None: ...
 
 class TextTestRunner:
     resultclass: _ResultClassType
-    def __init__(
-        self,
-        stream: TextIO | None = None,
-        descriptions: bool = True,
-        verbosity: int = 1,
-        failfast: bool = False,
-        buffer: bool = False,
-        resultclass: _ResultClassType | None = None,
-        warnings: type[Warning] | None = None,
-        *,
-        tb_locals: bool = False,
-    ) -> None: ...
+    stream: TextIO
+    descriptions: bool
+    verbosity: int
+    failfast: bool
+    buffer: bool
+    warnings: type[Warning] | None
```

Comment:
`self.warnings` is passed as the first argument of `warnings.simplefilter` here:

https://github.com/python/cpython/blob/6c484c39beeb66d40ef0a73cc4f1e900ea498cfa/Lib/unittest/runner.py#L232-L234

That implies that `warnings` should be `warnings._ActionKind | None`:

https://github.com/python/typeshed/blob/8888ac0d6dcb2d20b1621550cb64b69eba289f11/stdlib/warnings.pyi#L38

(The same goes for the annotation of the parameter in the `__init__` method -- the current type looks incorrect to me)

Reply:
I went with `warnings: str | None`, since it is has this type in `unittest/main.pyi` as well.

Reply:
OK